packages/devtools_app/lib/src/shared/table.dart

```diff
@@ -432,14 +459,15 @@ class TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
   void didUpdateWidget(TreeTable<T> oldWidget) {
     super.didUpdateWidget(oldWidget);
 
-    if (widget == oldWidget) return;
-
     cancelListeners();
 
     addAutoDisposeListener(selectionNotifier, () {
       setState(() {
         final node = selectionNotifier.value.node;
+        final index = selectionNotifier.value.flatListIndex as double;
+
         expandParents(node?.parent);
+        scrollToPosition(scrollController, index);
```

Comment:
we shouldn't need to add any of this if we add the `selectionNotifier` listener back to the base _TableState class.